malaria-app-android/src/main/java/com/peacecorps/malaria/ui/play/medicine_store/MedicineStoreFragment.java

```diff
@@ -0,0 +1,213 @@
+        presenter.attachView(this);
+        return view;
+    }
+
+    @Override
+    protected int getContentResource() {
+        return R.layout.medicine_store;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        init();
+    }
+
+    @Override
+    protected void init() {
+        presenter.checkMedicineStore();
+    }
+
+    @Override
+    public void displayMedicineStoreValues(String drugName, String days) {
+        medicineName.setText(drugName);
+        daysLeft.setText(days);
+    }
+
+    @OnClick(R.id.btn_add_medicine)
+    public void addMedicineListener(View view) {
+        final Dialog addMedicineDialog = new Dialog(context, android.R.style.Theme_DeviceDefault_Dialog_NoActionBar);
+        addMedicineDialog.setContentView(R.layout.add_medicine_dialog);
+
+        // Setting Dialog Title
+        final EditText medicineQuantityEt = addMedicineDialog.findViewById(R.id.add_medicine_quantity);
+        addMedicineDialog.findViewById(R.id.add_medicine_quantity).requestFocus();
+        addMedicineDialog.findViewById(R.id.btn_dialog_add_medicine).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (presenter.checkMedicineNumberValidity(medicineQuantityEt.getText().toString().trim())) {
+                    presenter.updateMedicineStoreValue(medicineQuantityEt.getText().toString().trim());
+                    addMedicineDialog.dismiss();
+                } else {
+                    medicineQuantityEt.setError("Quantity Required");
+                    medicineQuantityEt.requestFocus();
+                }
+            }
+        });
+        addMedicineDialog.findViewById(R.id.btn_dialog_cancel).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                addMedicineDialog.dismiss();
+            }
+        });
+
+        // Showing Alert Message
+        addMedicineDialog.show();
+    }
+
+    @OnClick(R.id.btn_order_medicine)
+    public void orderMedicineListener(View view) {
+        final Dialog orderMedicineDialog = new Dialog(context, android.R.style.Theme_DeviceDefault_Dialog_NoActionBar);
+        orderMedicineDialog.setContentView(R.layout.order_medicine_dialog);
+
+        // Setting Dialog Title
+        final EditText quantity = orderMedicineDialog.findViewById(R.id.btn_order_quantity);
+
+        orderMedicineDialog.findViewById(R.id.btn_order_quantity).requestFocus();
+
+        //implement the email button
+        orderMedicineDialog.findViewById(R.id.btn_order_by_email).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (presenter.checkMedicineNumberValidity(quantity.getText().toString().trim())) {
```

Comment:
Its good to avoid these long list of getters inside if statement, because, if fails not only on number is invalid, but also on failed access of quantity,

Reply:
Yeah, I don't understand the line ''also fails on access of quantity'.
Does it mean that quantity can be null & it will result in error itself? or something different.
Can u suggest something here? Confused.

Reply:
looks like, you will address this separately here:
https://github.com/systers/malaria-app-android/issues/350